[SPARK-12517] add default RDD name for one created via sc.textFile

wyaron opened the pull request:
The feature was first added at commit: 7b877b27053bfb7092e250e01a3b887e1b50a109 but was later removed (probably by mistake) at commit: fc8b58195afa67fbb75b4c8303e022f703cbf007. 
This change sets the default path of RDDs created via sc.textFile(...) to the path argument.

Here is the symptom:
- Using spark-1.5.2-bin-hadoop2.6:

scala> sc.textFile("/home/root/.bashrc").name
res5: String = null

scala> sc.binaryFiles("/home/root/.bashrc").name
res6: String = /home/root/.bashrc
- while using Spark 1.3.1:

scala> sc.textFile("/home/root/.bashrc").name
res0: String = /home/root/.bashrc

scala> sc.binaryFiles("/home/root/.bashrc").name
res1: String = /home/root/.bashrc

Commit:
add default RDD name for one create via sc.textFile

The feature was first added at commit: 7b877b27053bfb7092e250e01a3b887e1b50a109 but was later removed (probably by mistake) at at commit: fc8b58195afa67fbb75b4c8303e022f703cbf007.

here is the symptom:

using spark-1.5.2-bin-hadoop2.6 I get:
=================================
scala> sc.textFile("/home/root/.bashrc").name
res5: String = null

scala> sc.binaryFiles("/home/root/.bashrc").name
res6: String = /home/root/.bashrc

while using Spark 1.3.1:
=================================

scala> sc.textFile("/home/root/.bashrc").name
res0: String = /home/root/.bashrc

scala> sc.binaryFiles("/home/root/.bashrc").name
res1: String = /home/root/.bashrc

AmplabJenkins commented:
Can one of the admins verify this patch?

andrewor14 commented:
@wyaron please file a JIRA and attach to the title of this PR. See how other patches are opened.

andrewor14 commented:
The changes here look OK to me.

andrewor14 commented:
ok to test

SparkQA commented:
**[Test build #48271 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48271/consoleFull)** for PR 10456 at commit [`86efdad`](https://github.com/apache/spark/commit/86efdade980f63cce9f87696716c5d0c74eef245).

SparkQA commented:
**[Test build #48271 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48271/consoleFull)** for PR 10456 at commit [`86efdad`](https://github.com/apache/spark/commit/86efdade980f63cce9f87696716c5d0c74eef245).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/48271/
Test FAILed.

sarutak commented:
LGTM as well. Could you add testcases for this change?

wyaron committed:
add unit test for setting default RDD name to its path (SPARK-12517)

This change extends SparkContextSuite to verify that RDDs that are created using file paths have their
default name set to the path.

wyaron committed:
add unit test for setting default RDD name to its path (SPARK-12517)

This change extends SparkContextSuite to verify that RDDs that are created using file paths have their
default name set to the path.

wyaron committed:
Merge branch 'master' of https://github.com/wyaron/spark

SparkQA commented:
**[Test build #48305 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48305/consoleFull)** for PR 10456 at commit [`081a723`](https://github.com/apache/spark/commit/081a723c3e5604f161d6440a105b44cc2350480d).

SparkQA commented:
**[Test build #48305 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48305/consoleFull)** for PR 10456 at commit [`081a723`](https://github.com/apache/spark/commit/081a723c3e5604f161d6440a105b44cc2350480d).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/48305/
Test FAILed.

wyaron committed:
add unit test for setting default RDD name to its path (SPARK-12517)

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/48306/
Test FAILed.

SparkQA commented:
**[Test build #48307 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48307/consoleFull)** for PR 10456 at commit [`fe3d87d`](https://github.com/apache/spark/commit/fe3d87d5051b1ba57c9d7dd2e7718cfb0a1961ec).

SparkQA commented:
**[Test build #48307 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48307/consoleFull)** for PR 10456 at commit [`fe3d87d`](https://github.com/apache/spark/commit/fe3d87d5051b1ba57c9d7dd2e7718cfb0a1961ec).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/48307/
Test FAILed.

Commit:
restructure code to overcome 100 chars per line limit

SparkQA commented:
**[Test build #48309 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48309/consoleFull)** for PR 10456 at commit [`a2f5f07`](https://github.com/apache/spark/commit/a2f5f0733b019e036ca1ea2ac30dac53620310fe).

SparkQA commented:
**[Test build #48309 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48309/consoleFull)** for PR 10456 at commit [`a2f5f07`](https://github.com/apache/spark/commit/a2f5f0733b019e036ca1ea2ac30dac53620310fe).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/48309/
Test FAILed.

wyaron committed:
fix compile error in test unit

SparkQA commented:
**[Test build #48310 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48310/consoleFull)** for PR 10456 at commit [`50cd119`](https://github.com/apache/spark/commit/50cd1190d21c3721dc94d12288b677b7ef8276cd).

SparkQA commented:
**[Test build #48310 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48310/consoleFull)** for PR 10456 at commit [`50cd119`](https://github.com/apache/spark/commit/50cd1190d21c3721dc94d12288b677b7ef8276cd).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/48310/
Test FAILed.

wyaron committed:
yet another compile and style fix for unit test

Commit:
Merge branch 'master' of git://github.com/apache/spark

SparkQA commented:
**[Test build #48312 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48312/consoleFull)** for PR 10456 at commit [`825f3f3`](https://github.com/apache/spark/commit/825f3f3d38128b520544de690fc8c04605ffa51c).

SparkQA commented:
**[Test build #48313 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48313/consoleFull)** for PR 10456 at commit [`99b5bde`](https://github.com/apache/spark/commit/99b5bde4a3a949f5c0075016504936823bcbf5ea).

sarutak commented on core/src/test/scala/org/apache/spark/SparkContextSuite.scala:
```diff
@@ -274,6 +274,31 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext {
+
+      var targetPath = mockPath + "textFile"
+      assert(sc.textFile(targetPath).name == targetPath)
+
```
Could you compare by `===` instead of `==` ?
We can get better error messages by `===` operator.

sarutak commented on core/src/test/scala/org/apache/spark/SparkContextSuite.scala:
```diff
@@ -274,6 +274,31 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext {
+    test("Default path for file based RDDs is properly set (SPARK-12517)") {
+      sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local"))
+
+      // Test  filetextFile, wholeTextFiles, binaryFiles, hadoopFile and
```
Extra white space here.

SparkQA commented:
**[Test build #48312 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48312/consoleFull)** for PR 10456 at commit [`825f3f3`](https://github.com/apache/spark/commit/825f3f3d38128b520544de690fc8c04605ffa51c).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/48312/
Test FAILed.

sarutak commented on core/src/test/scala/org/apache/spark/SparkContextSuite.scala:
```diff
@@ -274,6 +274,31 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext {
     }
   }
 
+    test("Default path for file based RDDs is properly set (SPARK-12517)") {
```
Indentation is broken.

SparkQA commented:
**[Test build #48313 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48313/consoleFull)** for PR 10456 at commit [`99b5bde`](https://github.com/apache/spark/commit/99b5bde4a3a949f5c0075016504936823bcbf5ea).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/48313/
Test FAILed.

wyaron committed:
also fix default path for sc.wholeTextFiles and identation based on CR

wyaron commented:
Thanks for the quick CR !
Apparently, the new unit test has just failed for sc.wholeTextFiles(...) :-)
My guess is that it was broken in commit 297048ff730c022397e03afaf5ec9c91c811102a. 
The change called setName on the WholeTextFileRDD and then called map on it which yields a new RDD without the default name set (see code snippet below).

It seems that setName should be called AFTER the map is invoked, hence on the *returned* RDD.
Note that wholeTextFiles provides the file name for each data item in the RDD so the defaul path name is less useful... still, for consistency purposes it should be fixed. 

here is the original change:
  new WholeTextFileRDD(
      this,
      classOf[WholeTextFileInputFormat],
      - classOf[String],
      - classOf[String],
      + classOf[Text],
      + classOf[Text],
      updateConf,
      - minPartitions).setName(path)
      + minPartitions).setName(path).map(record => (record._1.toString, record._2.toString))
  }

I have committed the proposed change which moves setName after the map. I've tested locally and now the name field is valid. 

does it make sense ?

SparkQA commented:
**[Test build #48314 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48314/consoleFull)** for PR 10456 at commit [`08d3fe2`](https://github.com/apache/spark/commit/08d3fe25e4621812d963af502815a508da8930c8).

SparkQA commented:
**[Test build #48314 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/48314/consoleFull)** for PR 10456 at commit [`08d3fe2`](https://github.com/apache/spark/commit/08d3fe25e4621812d963af502815a508da8930c8).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/48314/
Test PASSed.

sarutak commented:
> I have committed the proposed change which moves setName after the map. I've tested locally and now the name field is valid.
> 
> does it make sense ?

Yeah, I think it's good catch.

andrewor14 commented:
LGTM

sarutak commented:
O.K, I'll merge this into `master` and `branch-1.6`. Thanks @wyaron !
